R/estimateAlpha.R

```diff
@@ -0,0 +1,274 @@
+                                              args.fun=list(index=index[i], assay.type="subsampled"),
+                                              ...,
+                                              name=name[i])
+                  } else {
+                      # Estimate index without rarefaction
+                      # warning is supressed due to the deprication of the functions called.
+                      suppressWarnings(x <- do.call(FUN, args = c(list(x, assay.type=assay.type,
+                                                                       index=index[i],
+                                                                       name=name[i]),
+                                                                  list(...))))
+                  }
+              }
+              return(x)
+    }
+)
+## Helper functions
+
+.get_indices <- function(
+        measure) {
+    switch(measure,
+           "diversity" = c("coverage_diversity", "faith_diversity", 
+                           "fisher_diversity", "gini_simpson_diversity",
+                           "inverse_simpson_diversity",
+                           "log_modulo_skewness_diversity", "shannon_diversity"),
+           "dominance" = c("absolute_dominance",
+                           "dbp_dominance", "core_abundance_dominance",
+                           "gini_dominance", "dmn_dominance", "relative_dominance",
+                           "simpson_lambda_dominance"),
+           "evenness" = c("camargo_evenness", "pielou_evenness", "simpson_evenness",
+                          "evar_evenness", "bulla_evenness"),
+           "richness" = c("ace_richness", "chao1_richness", "hill_richness", 
+                          "observed_richness"))
+}
+
+.alpha_rarefaction <- function(
+        x,
+        n.iter=1L,
+        args.sub=list(assay.type="counts", min_size=min(colSums(assay(x, "counts")),
+                                                        na.rm = TRUE),
+                      verbose=FALSE),
+        FUN=.estimate_diversity,
+        args.fun=c(index="shannon",
+                   assay.type="subsampled"),
+        ...,
+        name = args.fun$index) {
+    # Calculating the mean of the subsampled alpha estimates ans storing them
+    colData(x)[, name] <- lapply(seq(n.iter), function(j){
+        # subsampling the counts from the original tse object
+        x_sub <- do.call(subsampleCounts, args = c(list(x), args.sub))
+        # calculating the diversity indices on the subsampled object
+        # warnings are supressed due to the depricated warning of the alpha
+        # measure functions
+        suppressWarnings(x_sub <- do.call(FUN, args = c(list(x_sub),
+                                                        args.fun,
+                                                        list(...))))
+        # Storing estimate results
+        colData(x_sub)[, args.fun$index, drop=FALSE]
+    }) %>% data.frame(.) %>% rowMeans(.) %>% data.frame(.)
```

Comment:
You could split these to multiple steps. I mean first you could calculate the value to variable a, then do these modifications to it and then return

Reply:
?
Is this not working?

Reply:
Yes, it is working. However, it is little bit hard to follow. I am thinking if we need to debug things, so it is easier that thinkgs are done in steps rsther than with one go

Reply:
I do think it's easy to follow and it's straight forward. What is complicated there at the end of the day?

Reply:
it is perfectly fine in my opinion, you could perhaps change i tin future fixes and commits or what so ever.. :)

Reply:
I do think that using dplyr grammar for coding is already clear...

Reply:
As I said, this is good like this, but we have preferred base grammar, no big deal